server/src/main/java/org/spine3/server/entity/storage/QueryParameters.java

```diff
@@ -0,0 +1,184 @@
+     * Iterates over the Query parameters.
+     *
+     * <p>The iteration is performed synchronously to guarantee no unexpected behaviour when
+     * performing closure-based information within the {@code consumer}.
+     *
+     * @param consumer the {@link ParameterConsumer} processing each parameter separately.
+     */
+    public void forEach(ParameterConsumer consumer) {
+        for (Map.Entry<QueryOperator, Map<Column<?>, Object>> param : parameters.entrySet()) {
+            final QueryOperator operator = param.getKey();
+            applyConsumer(consumer, operator, param.getValue());
+        }
+    }
+
+    private static void applyConsumer(ParameterConsumer consumer,
+                                      QueryOperator operator,
+                                      Map<Column<?>, Object> paramValues) {
+        for (Map.Entry<Column<?>, Object> parameterValue : paramValues.entrySet()) {
+            consumer.consume(operator, parameterValue.getKey(), parameterValue.getValue());
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        QueryParameters that = (QueryParameters) o;
+        return Objects.equal(parameters, that.parameters);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(parameters);
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * The builder for the {@code QueryParameters}.
+     */
+    public static class Builder {
+
+        private final Map<QueryOperator, Map<Column<?>, Object>> parameters
+                = new EnumMap<>(QueryOperator.class);
+
+        private Builder() {
+            // Prevent direct initialization
+        }
+
+        /**
+         * Put the Query parameter represented by the arguments into the resulting instance of
+         * {@code QueryParameters}.
+         *
+         * @param operator the parameter comparison operator
+         * @param column   the parameter target {@link Column}
+         * @param value    the parameter value
+         * @return self for method chaining
+         */
+        public Builder put(QueryOperator operator, Column<?> column, @Nullable Object value) {
+            checkNotNull(operator);
+            checkNotNull(column);
+
+            Map<Column<?>, Object> params = parameters.get(operator);
+            if (params == null) {
+                params = new HashMap<>();
+                params.put(column, value);
+                parameters.put(operator, params);
+            } else {
+                params.put(column, value);
+            }
+            return this;
+        }
+
+        public QueryParameters build() {
+            return new QueryParameters(parameters);
+        }
+    }
+
+    /**
+     * An interface simplifying the iteration over the Query parameters.
```

Comment:
Please add more details to this description. At the moment it is not clear why _we_ have it, aside from some "simplification". The core committers and SPI users should understand what it is and what it does. Mentioning its usages and description of its method don't help at all.